services/src/main/java/org/exoplatform/addons/gamification/service/ChallengeService.java

```diff
@@ -88,4 +100,24 @@ public interface ChallengeService {
    *           ended yet
    */
   void deleteChallenge(Long challengeId, String username) throws IllegalAccessException, ObjectNotFoundException;
+
+  /**
+   * Search news by term
+   *
+   * @param term term
+   * @param offset Offset
+   * @param limit Limit
+   * @param username Username accessing challenge
+   * @throws IllegalAccessException when user is not authorized to delete
+   *           challenges
+   * @throws ObjectNotFoundException challenge not found
+   * @throws IllegalArgumentException when challenge has announcements or did not
+   *           ended yet
+   */
+  List<Challenge> search(String term, int offset, int limit, String username);
+
+  /**
+   * Retrieves all challenges .
+   **/
+  List<Challenge> getAllChallenges();
```

Comment:
we do not need to have an API that does not query results with pagination and limit